scipy/signal/signaltools.py

```diff
@@ -1020,8 +1020,9 @@ def convolve2d(in1, in2, mode='full', boundary='fill', fillvalue=0):
     val = _valfrommode(mode)
     bval = _bvalfromboundary(boundary)
 
-    with warnings.catch_warnings():
-        warnings.simplefilter('ignore', np.ComplexWarning)
+    with suppress_warnings() as sup:
```

Comment:
This is also non-test usage of suppress_warnings, which perhaps should be avoided.